a json data cause a panic?

```
panic: runtime error: slice bounds out of range

goroutine 1 [running, locked to thread]:
fyne.io/fyne/widget.(*textProvider).lineSizeToColumn(0xc001396380, 0xfffffffffffffff7, 0x6, 0x0, 0x0)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne@v1.2.1/widget/text.go:200 +0x2b4
fyne.io/fyne/widget.(*entryRenderer).moveCursor(0xc000143680)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne@v1.2.1/widget/entry.go:132 +0xdf
fyne.io/fyne/widget.(*Entry).pasteFromClipboard(0xc000185100, 0x130e260, 0xc000006018)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne@v1.2.1/widget/entry.go:497 +0x29b
fyne.io/fyne/widget.(*Entry).registerShortcut.func3(0x13064c0, 0xc0002fa1a0)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne@v1.2.1/widget/entry.go:1084 +0x79
fyne.io/fyne.(*ShortcutHandler).TypedShortcut(0xc000185158, 0x13064c0, 0xc0002fa1a0, 0x11ea000)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne@v1.2.1/shortcut.go:20 +0x10c
fyne.io/fyne/widget.(*Entry).TypedShortcut(0xc000185100, 0x13064c0, 0xc0002fa1a0, 0x12140b00)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne@v1.2.1/widget/entry.go:962 +0x56
fyne.io/fyne/internal/driver/glfw.(*window).keyPressed(0xc000072140, 0xc0000b2000, 0x56, 0x2f, 0x1, 0x2)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne@v1.2.1/internal/driver/glfw/window.go:978 +0x6be
github.com/go-gl/glfw/v3.2/glfw.goKeyCB(0x488e0d0, 0x2f00000056, 0x200000001)
	C:/Users/Administrator/go/pkg/mod/github.com/go-gl/glfw@v0.0.0-20181213070059-819e8ce5125f/v3.2/glfw/input.go:288 +0x7f
github.com/go-gl/glfw/v3.2/glfw._cgoexpwrap_861c8851bbe2_goKeyCB(0x488e0d0, 0x2f00000056, 0x200000001)
	_cgo_gotypes.go:2094 +0x52
github.com/go-gl/glfw/v3.2/glfw._Cfunc_glfwPollEvents()
	_cgo_gotypes.go:1156 +0x48
github.com/go-gl/glfw/v3.2/glfw.PollEvents()
	C:/Users/Administrator/go/pkg/mod/github.com/go-gl/glfw@v0.0.0-20181213070059-819e8ce5125f/v3.2/glfw/window.go:777 +0x27
fyne.io/fyne/internal/driver/glfw.(*gLDriver).runGL(0xc00031c030)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne@v1.2.1/internal/driver/glfw/loop.go:84 +0x301
fyne.io/fyne/internal/driver/glfw.(*gLDriver).Run(0xc00031c030)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne@v1.2.1/internal/driver/glfw/driver.go:66 +0x32
fyne.io/fyne/internal/driver/glfw.(*window).ShowAndRun(0xc000072140)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne@v1.2.1/internal/driver/glfw/window.go:367 +0x76
```


some  strings include Tab character copy to `widget.NewMultiLineEntry()`

my data:
```
{
	"const": {
		"font": "./source/font.ttf",
		"theme": "light",
		"scale": 1
	}
}
```

--- a/widget/entry.go
+++ b/widget/entry.go
@@ -490,8 +490,13 @@ func (e *Entry) pasteFromClipboard(clipboard fyne.Clipboard) {
 		e.CursorColumn += len(runes)
 	} else {
 		e.CursorRow += newlines
-		lastNewline := strings.LastIndex(text, "\n")
-		e.CursorColumn = len(runes) - lastNewline - 1
+		lastNewlineIndex := 0
+		for i, r := range runes {
+			if r == '\n' {
+				lastNewlineIndex = i
+			}
+		}
+		e.CursorColumn = len(runes) - lastNewlineIndex - 1
 	}
 	e.updateText(provider.String())
 	Renderer(e).(*entryRenderer).moveCursor()

--- a/widget/entry_test.go
+++ b/widget/entry_test.go
@@ -1441,3 +1441,21 @@ func TestEntry_PageUpDown(t *testing.T) {
 		assert.Equal(t, 7, e.CursorColumn)
 	})
 }
+
+func TestEntry_PasteUnicode(t *testing.T) {
+	e := NewMultiLineEntry()
+	e.SetText("line")
+	e.CursorColumn = 4
+
+	clipboard := test.NewClipboard()
+	clipboard.SetContent("thing {\n\titem: 'val测试'\n}")
+	shortcut := &fyne.ShortcutPaste{Clipboard: clipboard}
+	handled := e.TypedShortcut(shortcut)
+
+	assert.True(t, handled)
+	assert.Equal(t, "thing {\n\titem: 'val测试'\n}", clipboard.Content())
+	assert.Equal(t, "linething {\n\titem: 'val测试'\n}", e.Text)
+
+	assert.Equal(t, 2, e.CursorRow)
+	assert.Equal(t, 1, e.CursorColumn)
+}

